File: app/src/main/java/com/neome/java/api/meta/base/dto/EntVdGeneratePaymentLink.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.EnumDefnPaymentMethodKind;
import com.neome.java.api.meta.base.Types.EnumPaymentProviderKind;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class EntVdGeneratePaymentLink extends EntVdAutoStepWithError
{
  @Nullable
  public EnumDefnPaymentMethodKind[] allowedPaymentMethodSet;

  @Nullable
  public StudioBuildArgBinder amount;

  @Nullable
  public StudioBuildArgBinder currency;

  @Nullable
  public StudioBuildArgBinder expiryDuration;

  @Nullable
  public StudioDtoArgValueParameter outputField;

  @Nullable
  public StudioBuildArgBinder paymentDescription;

  @Nullable
  public EnumPaymentProviderKind paymentProvider;

  @Nullable
  public StudioDtoArgValueParameter referenceIdField;

  @Nullable
  public StudioDtoArgValueParameter spreadsheetRowIdField;
}

File: app/src/main/java/com/neome/api/meta/base/dto/EntVdGeneratePaymentLink.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnPaymentMethodKind
import com.neome.api.meta.base.Types.EnumPaymentProviderKind

interface EntVdGeneratePaymentLink : EntVdAutoStepWithError {
    val allowedPaymentMethodSet: List<EnumDefnPaymentMethodKind>?
    val amount: StudioBuildArgBinder?
    val currency: StudioBuildArgBinder?
    val expiryDuration: StudioBuildArgBinder?
    val outputField: StudioDtoArgValueParameter?
    val paymentDescription: StudioBuildArgBinder?
    val paymentProvider: EnumPaymentProviderKind?
    val referenceIdField: StudioDtoArgValueParameter?
    val spreadsheetRowIdField: StudioDtoArgValueParameter?
}
